From 35dffd4f4a82a18897bf0b78df8f1402ef5e2d03 Mon Sep 17 00:00:00 2001
From: Josua Mayer <josua@solid-run.com>
Date: Thu, 31 Oct 2024 17:35:21 +0100
Subject: [PATCH] lx2160acex7: clearfog-cx: configure qsfp ports type phy

Interface type PHY allows Linux to configure ethernet speed at runtime
for each port, rather than sticking to assignment from serdes protocols.

This is particularly useful with SD1 protocol 18 which by default drives
just two ports at 25Gbps while QSFP connector has 4.
At protocol 18 Linux can switch any of the 8 ports between 10Gbps and
25Gbps as needed.

Due to lack of software support, the QSFP ports need to define speed in
device-tree.

Signed-off-by: Josua Mayer <josua@solid-run.com>
---
 config/lx2160a/LX2160A-CEX7/clearfog-cx-s1_8-s2_0-dpc.dts | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/config/lx2160a/LX2160A-CEX7/clearfog-cx-s1_8-s2_0-dpc.dts b/config/lx2160a/LX2160A-CEX7/clearfog-cx-s1_8-s2_0-dpc.dts
index dcc376e..9a508d2 100644
--- a/config/lx2160a/LX2160A-CEX7/clearfog-cx-s1_8-s2_0-dpc.dts
+++ b/config/lx2160a/LX2160A-CEX7/clearfog-cx-s1_8-s2_0-dpc.dts
@@ -69,19 +69,19 @@
 	board_info {
 		ports {
 			mac@3 {
-				link_type = "MAC_LINK_TYPE_FIXED";
+				link_type = "MAC_LINK_TYPE_PHY";
 			};
 
 			mac@4 {
-				link_type = "MAC_LINK_TYPE_FIXED";
+				link_type = "MAC_LINK_TYPE_PHY";
 			};
 
 			mac@5 {
-				link_type = "MAC_LINK_TYPE_FIXED";
+				link_type = "MAC_LINK_TYPE_PHY";
 			};
 
 			mac@6 {
-				link_type = "MAC_LINK_TYPE_FIXED";
+				link_type = "MAC_LINK_TYPE_PHY";
 			};
 
 			mac@7 {
-- 
2.43.0

